js/sagas/onboarding.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * A saga that manages the Onboarding.
+ *
+ * @flow
+ */
+
+import { type Saga } from 'redux-saga'
+import { takeLatest, select, take, call, put } from 'redux-saga/effects'
+import { NavigationActions } from 'react-navigation'
+
+import { type GlobalState } from '../reducers/types'
+import {
+  SESSION_INITIALIZE_SUCCESS,
+  TOS_ACCEPT_REQUEST,
+  TOS_ACCEPT_SUCCESS,
+  PIN_CREATED_REQUEST
+} from '../store/actions/constants'
+import ROUTES from '../navigation/routes'
+
+// The Pin step of the Onboarding
+function* pinStep(): Saga<void> {
+  // From the state check ir the user already created a Pin
+  const isPinCreated: boolean = yield select(
```

Comment:
it looks like if isPinCreated == true this saga does nothing. Can we move this check at the caller level so we handle only the isPinCreated == false branch here ?

Reply:
It is still incomplete because i have to still code the next screens. The condition `isPinCreated == true` will bring the user to the next step.

Reply:
can we split the saga and move it at the caller level ?

ie.

isPinCreated -> run <saga after pin creation>
!isPinCreated -> run pinStep

Reply:
The problem is 'the caller level'. I mean we said is better to start the pinSaga listening to an action. We hypothesized to use take([TOS_ACCEPT_SUCCESS, TOS_ALREADY_ACCEPTED], pinStep) to start the pinStep, the dispatch occur in the tosStep. Now, correct me if wrong, you want to skip the pinStep saga if not nedded, but the question is: where we check isPinCreated? Something like pinSaga (the new parent saga) -> isPinCreated ? 'DISPATCH PIN_ALREADY_CREATED' : pinCreationSaga (the old pinStep with only the !isPinCreated branch)?

Reply:
I see your point. I'll think about this :) In the meantime it's ok the approach you suggest, just move the selectors (when needed) outside sagas.

ie. use yield select(isPinRequired) where isPinRequired is a selector function that is defined somewhere in order to be used outside sagas as well.